modules/widgets/top-posts.php

```diff
@@ -105,6 +130,14 @@ function update( $new_instance, $old_instance ) {
 			$instance['count'] = 10;
 		}
 
+		$allowed_post_types = array_values( get_post_types( array( 'public' => true ) ) );
+		$instance['types'] = $new_instance['types'];
+		foreach( $new_instance['types'] as $type ) {
+			if ( ! in_array( $type, $allowed_post_types ) ) {
+				unset( $type );
```

Comment:
I'm not sure I understand, don't we want to remove the whole `$type` if it is not allowed?